src/log.c

```diff
@@ -206,24 +145,42 @@ static int writeEntry(Log *log, raft_entry_t *ety)
     size_t offset = FileSize(&log->file);
     size_t idxoffset = FileSize(&log->idxfile);
 
-    off += writeLength(buf + off, sizeof(buf) - off, '*', ENTRY_ELEM_COUNT);
-    off += writeString(buf + off, sizeof(buf) - off, ENTRY_STR);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->term);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->id);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->type);
-    off += writeLength(buf + off, sizeof(buf) - off, '$', ety->data_len);
+    /* header */
+    off = generateEntryHeader(ety, buf, sizeof(buf));
+    if (FileWrite(&log->file, buf, off) != (ssize_t) off) {
+        goto error;
+    }
 
-    if (FileWrite(&log->file, buf, off) != (ssize_t) off ||
-        FileWrite(&log->file, ety->data, ety->data_len) != ety->data_len ||
+    /* data */
+    if (FileWrite(&log->file, ety->data, ety->data_len) != ety->data_len ||
         FileWrite(&log->file, "\r\n", 2) != 2) {
         goto error;
     }
 
+    /* crc */
+    /* accumulating crc, so start with current crc */
+    long crc = crc16_ccitt(log->current_crc, buf, off);
+    crc = crc16_ccitt(crc, ety->data, ety->data_len);
+    crc = crc16_ccitt(crc, "\r\n", 2);
+
+    /* write crc as added element */
+    off = 0;
+    off += writeLong(buf + off, sizeof(buf) - off, crc);
```

Comment:
```suggestion
    off = writeLong(buf, sizeof(buf), crc);
```

Can we simplify a bit like this?

Reply:
ok, the reason I did it the same was to enable copy pasting into other places less error prone, but changed it.